src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/KeyedDependencyInjectionSpecificationTests.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Xunit;
+using Microsoft.Extensions.DependencyInjection.Specification.Fakes;
+using System.Linq;
+
+namespace Microsoft.Extensions.DependencyInjection.Specification
+{
+    public abstract partial class KeyedDependencyInjectionSpecificationTests
+    {
+        protected abstract  IServiceProvider CreateServiceProvider(IServiceCollection collection);
+
+        [Fact]
+        public void ResolveKeyedService()
+        {
+            var service1 = new Service();
```

Comment:
style nit\preference: I favor `Service service1 = new()` syntax and try to avoid `var` in general whenever the type isn't spelled out later in the same line, but it looks like DI uses `var` quite a bit so keeping consistency at least within each file is good. Also, consistency is not so important for tests, but at least in production code.